client/src/components/AddScheduleForm.tsx

```diff
@@ -0,0 +1,204 @@
+            return 100;
+          }
+          return prevProgress + 25;
+        });
+      }, 1000);
+    }
+    return () => clearInterval(timer);
+  }, [isLoading]);
+
+  function handleCheckedChange(position: number) {
+    const updatedCheckedState = checkedState.map((item, index) =>
+      index === position ? !item : item
+    );
+    setCheckedState(updatedCheckedState);
+  }
+
+  async function handleSubmit(e: React.FormEvent) {
+    setIsLoading(true);
+    setProgress(0);
+    e.preventDefault();
+    try {
+      if (Number(timesPerDay) !== 0) {
+        const daysAdded: string[] = [];
+        for (let i = 0; i < checkedState.length; i++) {
+          if (checkedState[i]) daysAdded.push(days[i].label);
+        }
+        const newSchedule = {
+          medicationId: medication.id,
+          timesPerDay: timesPerDay,
+          daysOfWeek: daysAdded,
+          userId: 1,
+        };
+
+        const response = await fetch('/api/medications/schedule', {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify(newSchedule),
+        });
+        if (!response.ok)
+          throw new Error(`Response status: ${response.status}`);
+      }
+    } catch (error) {
+      setError(error);
+    } finally {
+      setTimeout(() => {
+        setIsLoading(false);
+        setCheckedState(new Array(days.length).fill(false));
+        setTimesPerDay('');
+        onScheduleComplete(medication);
+      }, 4000);
+    }
+  }
+
+  if (error) {
+    return (
+      <>
+        <p>{`Error: ${error}`}</p>
+      </>
+    );
+  }
+  return (
+    <>
+      <section className="container pt-4">
+        <Card className={`relative pt-4 ${isLoading && 'opacity-0'}`}>
+          <CardHeader className="text-2xl text-redblack">
+            <CardTitle>{medication.name}</CardTitle>
+            <CardDescription>{`Add ${medication.name} to your schedule`}</CardDescription>
+          </CardHeader>
+          <CardContent>
+            <form action="" onSubmit={(e) => handleSubmit(e)}>
+              <ul>
+                {days.map((day, index) => (
+                  <li key={day.id} className="flex items-center gap-2 mb-3">
+                    <Checkbox
+                      id={day.id}
+                      checked={checkedState[index]}
+                      onCheckedChange={() => handleCheckedChange(index)}
+                    />
+                    <label
+                      htmlFor={day.id}
+                      className="text-lg font-medium leading-none text-redblack peer-disabled:cursor-not-allowed peer-disabled:opacity-70">
+                      {day.label}
+                    </label>
+                  </li>
+                ))}
+              </ul>
+              <div className="flex flex-col gap-4 pt-4">
+                <label htmlFor="" className="text-lg text-redblack">
+                  How many doses do you take per day?
+                </label>
```

Comment:
If you don't connect a `label` to a form control via `for` or `htmlFor` in this case, the `label` isn't needed. You only need a `label` if you are connecting it to the form control.